test/integration/inviting_users_test.rb

```diff
@@ -384,4 +384,132 @@ class InvitingUsersTest < ActionDispatch::IntegrationTest
       end
     end
   end
+
+  context "with JavaScript enabled" do
+    setup do
+      use_javascript_driver
+
+      create(:organisation, name: "ABCDEF")
+      create(:organisation, name: "GHIJKL")
+      @organisation = create(:organisation, name: "MNOPQR")
+      create(:organisation, name: "STUVWX")
+      create(:organisation, name: "YZ1234")
+
+      superadmin = create(:superadmin_user)
+      visit root_path
+      signin_with(superadmin)
+
+      visit new_user_invitation_path
+      fill_in "Name", with: "H from Steps"
+      fill_in "Email", with: "h@from.steps"
+      select "Superadmin", from: "Role"
+
+      @autocomplete_input_element = find(".autocomplete__input")
+      @select_element = find("#user_organisation_id-select", visible: false)
+    end
+
+    should "be able to invite a user" do
+      assert_select_with_autocomplete(
+        autocomplete_input_element: @autocomplete_input_element,
+        select_element: @select_element,
+        option_text: @organisation.name,
+        option_value: @organisation.id.to_s,
+        unique_partial_string: "MNO",
+      )
+
+      click_button "Create user and send email"
+
+      new_user = User.find_by(email: "h@from.steps", role: Roles::Superadmin.name)
+      assert_not_nil new_user
+      assert_equal new_user.organisation, @organisation
+    end
+
+    should("not show an 'Add' button") { assert_no_selector "button", text: "Add" }
+
+    should "reset the value of the select element when it no longer matches what's shown in the autocomplete input" do
```

Comment:
How would you feel about extracting these tests into more "component" style tests? That way we'd only need to test it once as we're reusing the component across multiple pages.

As it stands we're testing this behaviour twice across the two integration tests. We could then restrict the two test files modified in this PR to only test a user inputting some text into the autocomplete and inviting a user.